packages/app/src/app/pages/Dashboard/Content/routes/Settings/TeamSettings/RegistryPreset/UsernamePassword.tsx

```diff
@@ -0,0 +1,68 @@
+import { Input } from '@codesandbox/components';
+import * as React from 'react';
+import { CustomFormField } from '../RegistryForm';
+
+type UsernamePasswordProps = {
+  setAuthKey: (key: string) => void;
+  disabled: boolean;
+  authKey: string;
+};
+
+export const UsernamePassword = ({
+  disabled,
+  setAuthKey,
+  authKey,
+}: UsernamePasswordProps) => {
+  let [initialUsername, initialPassword] = Buffer.from(
+    String(authKey),
+    'base64'
+  )
+    .toString('utf-8')
+    .split(':');
+
+  if (!initialPassword || !authKey.endsWith('=')) {
+    // Meaning that we parsed wrong
+    initialUsername = '';
+    initialPassword = '';
+  }
+
+  const [username, setUsername] = React.useState(initialUsername);
+  const [token, setToken] = React.useState(initialPassword);
+
+  React.useEffect(() => {
+    if (!username || !token) {
+      return;
+    }
+
+    // First unencode the token, this is what npm does as well
+    const unencodedToken = Buffer.from(token, 'base64').toString('utf-8');
+
+    const key = `${username}:${unencodedToken}`;
+    const base64Data = Buffer.from(key);
+    const newAuthKey = base64Data.toString('base64');
+    setAuthKey(newAuthKey);
+  }, [username, token, setAuthKey]);
```

Comment:
this hook will be called every time you input another character in the username or password fields. if that's not concerning for performance reasons, that's fine, otherwise I'd switch to an `onBlur` event handler and keep the Inputs uncontrolled with `defaultValue` instead of `value`

Reply:
Good catch!! I tested a bit more, and I think in terms of performance this is still fine, the conversion from base64 to utf-8 is fast enough, so I think we can keep this for now.